src/body.rs

```diff
@@ -117,6 +120,48 @@ fn mk_err<T>(_: T) -> Response {
     StatusCode::BAD_REQUEST.into_response()
 }
 
+pub struct MultipartFile(pub Vec<u8>);
+
+impl<S: 'static> Extract<S> for MultipartFile {
+    // Note: cannot use `existential type` here due to ICE
+    type Fut = FutureObj<'static, Result<Self, Response>>;
+
+    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+        // https://stackoverflow.com/questions/43424982/how-to-parse-multipart-forms-using-abonander-multipart-with-rocket
+        let ct = req
+            .headers()
+            .get("content-type")
+            .expect("no header content-type");
+        let ct = ct.to_str().expect("header content-type cannot to_str()");
+        let idx = ct.find("boundary=").expect("no boundary");
+        let boundary = ct[(idx + "boundary=".len())..].to_string();
+
+        let mut body = std::mem::replace(req.body_mut(), Body::empty());
+
+        FutureObj::new(Box::new(
+            async move {
+                let body = await!(body.read_to_vec()).map_err(mk_err)?;
+
+                let mut mp = Multipart::with_body(Cursor::new(body), boundary);
+                let mut file = None;
+                mp.foreach_entry(|mut entry| match entry.headers.name.as_str() {
```

Comment:
I'm a little confused about the intent here -- this looks like it's tied to only a very specific header for multipart uploads?  Probably we'll want to be more general.

Reply:
It might be worth just returning the multipart instead of `Vec<u8>` here and allow the users to loop through the response to find the file they are looking for.

Reply:
Yep, that is what I thought as well. I actually tired to just returning the multipart yesterday but I couldn't win the fight with the compiler. Perhaps I can try again and it would be nice if some of you could help me out :)